piglow/piglow_test.go

```diff
@@ -0,0 +1,260 @@
+package piglow_test
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/goiot/devices/piglow"
+	"golang.org/x/exp/io/i2c/driver"
+)
+
+type opener struct {
+	Buffer *bytes.Buffer
+}
+
+func (o opener) Open() (driver.Conn, error) {
+	return &conn{
+		Buffer: o.Buffer,
+	}, nil
+}
+
+type conn struct {
+	Buffer *bytes.Buffer
+}
+
+func (c conn) Tx(w, r []byte) error {
+	if w != nil {
+		if _, err := c.Buffer.Write(w); err != nil {
+			return err
+		}
+	}
+
+	if r != nil {
+		if _, err := c.Buffer.Read(r); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (conn) Close() error {
+	return nil
+}
+
+func openPiGlow(t *testing.T) (*piglow.PiGlow, *bytes.Buffer) {
+	o := opener{
+		Buffer: bytes.NewBuffer([]byte{}),
+	}
+	device, err := piglow.Open(o)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return device, o.Buffer
+}
+
+func assert(t *testing.T, expected, actual interface{}) {
+	if !reflect.DeepEqual(expected, actual) {
+		t.Fatal("expected", expected, "to equal", actual)
```

Comment:
Give some context about what is expected.

The idiomatic testing messages should be in the following form.

t.Fatalf("some context about the failed case; got = %v, want = %", got, want)